tests/ut/azure_iot_json_reader_ut.c

```diff
@@ -0,0 +1,582 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <setjmp.h>
+
+#include <cmocka.h>
+
+#include "azure_iot_json_reader.h"
+#include "azure_iot_mqtt.h"
+#include "azure_iot_hub_client.h"
+/*-----------------------------------------------------------*/
+
+/*
+ * {
+ * "property_one": "value_one",
+ * "property_two": "value_two"
+ * }
+ */
+#define azureiotjsonreaderutTEST_JSON \
+    "{\"property_one\":\"value_one\",\"property_two\":\"value_two\"}"
+
+/*
+ * {
+ * "property_one": "value_one"
+ * }
+ */
+#define azureiotjsonreaderutTEST_JSON_STRING \
+    "{\"property_one\":\"value_one\"}"
+
+/*
+ * {
+ * "property_one": {
+ *     "child_one":"value_one"
+ * },
+ * "property_two": "value_two"
+ * }
+ */
+#define azureiotjsonreaderutTEST_JSON_CHILDREN \
+    "{\"property_one\":{\"child_one\":\"value_one\"},\"property_two\":\"value_two\"}"
+
+/*
+ * {
+ * "property_one": true
+ * }
+ */
+#define azureiotjsonreaderutTEST_JSON_BOOL \
+    "{\"property\":true}"
+
+/*
+ * {
+ * "property_one": 42
+ * }
+ */
+#define azureiotjsonreaderutTEST_JSON_INT32 \
+    "{\"property\":42}"
+
+/*
+ * {
+ * "property_one": 42.42
+ * }
+ */
+#define azureiotjsonreaderutTEST_JSON_DOUBLE \
+    "{\"property\":42.42}"
+
+/*
+ * {
+ * "property_one": ["value_one", "value_two"],
+ * }
+ */
+#define azureiotjsonreaderutTEST_JSON_ARRAY \
+    "{\"property_one\":[\"value_one\",\"value_two\"]}"
+
+
+
+uint32_t ulGetAllTests();
+
+static void testAzureIoTJSONReader_Init_Failure( void ** ppvState )
+{
+    AzureIoTJSONReader_t xReader;
+    uint8_t * pucTestJSON = azureiotjsonreaderutTEST_JSON_STRING;
+
```

Comment:
if you create static const uint8_t data[] = "";
then here you can pass data, sizeof(data) - 1